JenkinsfileProductionRelease

```diff
@@ -17,36 +17,17 @@ node('heavy && linux && docker') {
   dockerUtils.withDocker(image: 'node:18', args: '-e HOME=/tmp -e NPM_CONFIG_PREFIX=/tmp/.npm -v /etc/passwd:/etc/passwd:ro') {
     stage('Setup') {
       sh 'npm ci'
-      sh 'npm i -g @salesforce/cli@2.27.6'
-    }
-    
-    stage('Npm publish') {
-      withCredentials([
-      string(credentialsId: 'NPM_TOKEN', variable: 'NPM_TOKEN')]) {
-        sh 'echo //registry.npmjs.org/:_authToken=$NPM_TOKEN > ~/.npmrc'
-        sh 'npm run promote:npm:latest'
-      }
-    }
-
-    stage('Quantic publish') {
-      withCredentials([
-        usernamePassword(credentialsId: 'github-commit-token', usernameVariable: 'GITHUB_USERNAME', passwordVariable: 'GITHUB_TOKEN'),
-        string(credentialsId: 'sfdx-auth-pkg-client-id', variable: 'SFDX_AUTH_CLIENT_ID'),
-        file(credentialsId: 'sfdx-auth-pkg-jwt-key', variable: 'SFDX_AUTH_JWT_KEY_FILE'),
-      ]) {
-        withEnv([
-          'SFDX_AUTH_JWT_USERNAME=sfdc.integration.devv2.hub@coveo.com'
-        ]) {
-          sh 'npx --no-install nx run quantic:"promote:sfdx:ci"'
-        }
-      }
     }
 
-    stage('Notify Docs') {
+    stage('Approve production release') {
```

Comment:
So, when we run deployment pipeline:
* When it reaches prod, we will "approve" through spinnaker the deploy to prod.
* This will run this script which will then "approve" on github.
Is that right ?

Can you explain more what the second step entails ?

Reply:
Yes, essentially!
Right now (before this PR), when Spinnaker finishes the stg check, it will release into the prod environment.
Part of the "release to prod environment" is the execution of JenkinsFileProduction.
Today, JenkinsFileProduction promote stuff into npm, SFDC and notify the doc.

What changes is what JenkinsFileProduction is doing: Instead of doing the job itself, it approves the usage of the Production GitHub environment.
This approval will allow the release workflow to continue through the `promote-prod` job.
This job then do what JenkinsFileProduction was doing (or failing to do for SFDX): Promote stuff into npm, SFDC and notify the doc